### DISCUSSION TITLE
Surface scattering and BRDF at given heights inside the atmosphere

### DISCUSSION THREAD

**Discussion by Srheft** [eradiate/eradiate#350] [Q&A]
Hi Eradiate,

Another new user question: Could an experienced member please point me to a note or section of the code where Eradiate retrieves surface scattering and radiance at given heights inside the atmosphere? 

Thank you,
S.

**Comment by leroyvn** [comment]
Hi @Srheft, Eradiate can probably do this, but not with automated components.

The [`mradiancemeter`](https://eradiate.readthedocs.io/en/latest/rst/reference_api/generated/autosummary/eradiate.scenes.measure.MultiRadiancemeterMeasure.html) measure computes radiance estimates anywhere, so you can place sensors at any altitude and with any orientation you like.

Now, a point of terminology: what do you mean by "surface scattering and radiance"? The [volumetric path tracer integrator](https://eradiate.readthedocs.io/en/latest/rst/reference_api/generated/autosummary/eradiate.scenes.integrators.VolPathIntegrator.html)'s random walk can be stopped at specified orders of scattering using the `max_depth` parameter, but it cannot separate contributions transmitted to the surface, then transmitted after one bounce to the sensor. That would require a modification of the integrator.

**Comment by Srheft** [comment]
Thanks for the info. 
I was imagining using the software for a field measurement scenario in which the sensor is sitting at a given height and is calibrated to measure the radiance and it is pointed to the ground to measure the surface radiance. Then one wants to use the software to get the expected radiance at the height of the sensor for the given atmospheric and solar parameters and a range of surface reflectances. BOA reflectance in this scenario is an iterating input for the software to get the radiance at a given height to find a matching radiance from both the software and the sensor and this reveals the surface reflectance.  I should not have used the phrase scattering in my original message.
